lib/base/index.js

```diff
@@ -1,136 +1,138 @@
 // @flow
 import {keybaseExec, rmdirRecursive, randomTempDir, keybaseServiceStartup} from '../utils'
-import {withStore, actionInit, actionPreinit} from '../store'
-import type {Store} from '../store/types'
-import type {DeviceUsernamePair} from '../types'
-import type {Init, Deinit} from './types'
 
-/**
- * @ignore
- * getCurrentUsernameAndDevicename returns { username, devicename } from keybase status -j
- */
-const getCurrentUsernameAndDevicename = async (store: Store): Promise<DeviceUsernamePair> => {
-  const {homeDir} = store.getState()
-  const status = await keybaseExec(homeDir, ['status', '--json'], {
-    json: true,
-  })
-  if (status && status.Username && status.Device && status.Device.name) {
-    return {username: status.Username, devicename: status.Device.name}
-  } else {
-    throw new Error('failed to get username + device name')
-  }
-}
+export type Initialized = false | 'paperkey' | 'currentService'
 
-/**
- * @ignore
- * guardInitialized takes the current store and the api function to guard
- * It ensures that the following conditions remain true
- *  1. The bot was initialized before calling the given api function
- *  2. The bot has username and devicename if it has been initialized
- *  2. The current username and devicename have not changed values
- *
- * If any condition is false, an error is thrown
- * Otherwise, return the function (to be executed later)
- *
- */
-export const guardInitialized = async (store: Store): Promise<void> => {
-  const currentDPair = await getCurrentUsernameAndDevicename(store)
-  const {initialized, username, devicename} = store.getState()
-  if (
-    !initialized ||
-    !currentDPair ||
-    !username ||
-    !devicename ||
-    currentDPair.username !== username ||
-    currentDPair.devicename !== devicename
-  ) {
-    throw new Error('Uh-oh, username has changed or we never initialized correctly.')
-  }
-}
+export type InitOptions = {|
+  verbose?: boolean,
+  homeDir?: string,
+|}
+
+export type DeviceUsernamePair = {|username: string, devicename: string|}
 
-/**
- * @namespace bot
- */
+class BaseBot {
+  initialized: Initialized
+  username: void | string
+  devicename: void | string
+  homeDir: void | string
```

Comment:
`homeDir` can be `void` in the case where someone uses the standard home directory version of the service.